mula/scheduler/server/serializers/task.py

```diff
@@ -0,0 +1,44 @@
+import enum
+import uuid
+
+from pydantic import BaseModel, ConfigDict
+
+from .p_item import PrioritizedItem
+
+
+class TaskStatus(str, enum.Enum):
+    # Task has been created but not yet queued
+    PENDING = "pending"
+
+    # Task has been pushed onto queue and is ready to be picked up
+    QUEUED = "queued"
+
+    # Task has been picked up by a worker
+    DISPATCHED = "dispatched"
+
+    # Task has been picked up by a worker, and the worker indicates that it is
+    # running.
+    RUNNING = "running"
+
+    # Task has been completed
+    COMPLETED = "completed"
+
+    # Task has failed
+    FAILED = "failed"
+
+    # Task has been cancelled
+    CANCELLED = "cancelled"
+
+
+class Task(BaseModel):
```

Comment:
So these are mainly used as BaseModels that allow for patching and hence the default values? I also looked into this recently for the boefjes, where @ammar92 pointed to [the suggested naming convention](https://fastapi.tiangolo.com/tutorial/extra-models/) of, in this case `TaskIn`.

Reply:
https://github.com/minvws/nl-kat-coordination/pull/2944#issuecomment-2185777061